### DISCUSSION TITLE
Diagramme DATABASE

### DISCUSSION THREAD

**Discussion by NabilDjerroud** [582-41E-MA/projet-web-2-CollRen#17] [Remises]
diagramme DB

![image](https://github.com/582-41E-MA/projet-web-2-CollRen/assets/144846468/9376b861-c3db-40da-b443-6976acbd67c8)

**Comment by NabilDjerroud** [comment]
@claraquintela @CollRen voici le diagramme de la DB

**Reply by claraquintela** [reply]
Salut @NabilDjerroud 

Nous venons de parler à Marcos. il faut inclure le columne "description" dans la table "voiture".

STP, confirme que tu as lu cette msg :)

**Reply by claraquintela** [reply]
Il faut que ça soit un TEXTLONG  dans la DB 

:)

**Reply by NabilDjerroud** [reply]
Hello @claraquintela , cest fait ce soir je vais commencer avec figma je vous envoi la diagramme et le sql

**Reply by NabilDjerroud** [reply]
[sql-db.txt](https://github.com/user-attachments/files/15979712/sql-db.txt)

@claraquintela @CollRen bonjour, j'espère que vous allez bien, je vous mets en pièce jointe ici le fichier texte contenant le sql de la db avec la colonne description

**Comment by CollRen** [comment]
@NabilDjerroud @claraquintela 
Est-ce qu'on corriege une erreur dans le nom d'une colonne pour la table commandes? Ou on fait avec ?
expdedition_id 

Devrait se lire

expedition_id

**Comment by CollRen** [comment]
Je pense qu'on devrait la corriger parce que c'est une erreur qui s'est glissé d'une autre façon qu'avec notre projet.

Dans notre projet c'est écrit correctement... Je vais faire le changement dans mon php my admin et si ça créer des bugs je vous reviens. Sinon, faite aussi la correction.

**Comment by claraquintela** [comment]
Oui!! On la corrige avant de commencer les travaux :)


Em ter., 16 de jul. de 2024 às 13:09, René de Montigny <
***@***.***> escreveu:

**Comment by CollRen** [comment]
@NabilDjerroud @claraquintela 

Voici le lien de la base de données qui contient les taux de taxes ainsi que les villes ( et ville_id pour les utilisateurs)

https://drive.google.com/file/d/1aegTbmF3sLrY5I6uDGPsUBkqRZJgaSZJ/view?usp=drive_link

À plus,

**Reply by NabilDjerroud** [reply]
Super merci René

**Reply by NabilDjerroud** [reply]
@CollRen Il y a un souci avec la création d'une voiture, je n'arrive pas à créer une nouvelle !

**Reply by NabilDjerroud** [reply]
peux tu essayer de ton côté stp ?

**Comment by claraquintela** [comment]
Nabil, as tu changer ta DB comme Rene a suggeré?

Em qua., 17 de jul. de 2024 às 14:54, Nabil ***@***.***>
escreveu:

**Reply by NabilDjerroud** [reply]
oui j'ai changé avec celle qu'il l 'a envoyé en dernier

**Reply by claraquintela** [reply]
Pour moi, il fonctionne :)

**Reply by NabilDjerroud** [reply]
voici ce que j'ai comme erreur !

**Reply by NabilDjerroud** [reply]
![image](https://github.com/user-attachments/assets/8c4cc7a4-fd98-4ec6-bda6-72b15fffb04a)

**Reply by claraquintela** [reply]
Error 500 c'est relatif à ta base de donné. As tu vu ton fichier db config? ton fetch? etc etc?

De mon coté, tout est beau

**Reply by NabilDjerroud** [reply]
![image](https://github.com/user-attachments/assets/ea1d9106-b1fb-4959-b34a-3cbd85381d37)


![image](https://github.com/user-attachments/assets/53b6fe4a-409e-48f2-bfe3-cdf066e548ce)

**Reply by CollRen** [reply]
J'ai fait un console.log de modele_id à la ligne 123 de voitureCreate et c'est ce qui cause l'erreur. Je vais essayer de régler en t'envoyer un PR

**Reply by NabilDjerroud** [reply]
ok parfait René merci !

**Reply by CollRen** [reply]
OK il y a deux choses:

Dans la bd il faut aller dans voitures/index et changer model par modele dans la colonne 'nom de l'index' voir capture d'écran:

ET dans notre projet: model/voiture à la ligne de commande_id il faut mettre true dans allowNull

```js
        commande_id: {
            type: Sequelize.INTEGER,
            allowNull: true,
```

**Reply by CollRen** [reply]
<img width="1133" alt="Capture d’écran, le 2024-07-17 à 16 46 25" src="https://github.com/user-attachments/assets/b77097a9-a491-4e20-bcf1-756a870d368a">

**Reply by CollRen** [reply]
@NabilDjerroud @claraquintela  

Comment régler le Bug pour la création de voiture ci-haut:

**Reply by NabilDjerroud** [reply]
parfait merci René

**Reply by NabilDjerroud** [reply]
Tout est beau enfin, merci @CollRen et @claraquintela
